In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns


C:\Users\Connor\AppData\Local\Temp\ipykernel_1636\2259232837.py:1: DeprecationWarning: 
Pyarrow will become a required dependency of pandas in the next major release of pandas (pandas 3.0),
(to allow more performant data types, such as the Arrow string type, and better interoperability with other libraries)
but was not found to be installed on your system.
If this would cause problems for you,
please provide us feedback at https://github.com/pandas-dev/pandas/issues/54466
        
  import pandas as pd


#### Importing the Libaries I will need off the bat

# Importing the Data

In [3]:
df = pd.read_csv("Valorant_data_cleaned.csv")


# Starting some predictions useing models 

In [4]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.impute import SimpleImputer
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeRegressor
from sklearn.tree import plot_tree
from sklearn.svm import SVR

### Importing the Libaries I will possibly need for making my models

In [5]:
X = df[['rating', 'rating-t', 'rating-ct', 'Average combat score', 'kills', 'k-t', 'k-ct', 'deaths',
        'd-t', 'd-ct', 'assists', 'a-t', 'a-ct', 'total kills minus deaths', 'tkmd-t', 'tkmd-ct', 'kills assists survive trade perct', 'average damage per round', 'headshot percentage', 'first kill', 'fk-t', 'fk-ct', 'first death', 'fd-t', 'fd-ct']]

### Making X copnsisting of columns: 'rating', 'rating-t', 'rating-ct', 'Average combat score', 'kills', 'k-t', 'k-ct', 'deaths', 'd-t', 'd-ct', 'assists', 'a-t', 'a-ct', 'total kills minus deaths', 'tkmd-t',
### 'tkmd-ct', 'kills assists survive trade perct', 'average damage per round', 'headshot percentage', 'first kill', 'fk-t', 'fk-ct', 'first death', 'fd-t', 'fd-ct'

In [6]:
y = df['Team 1 Win']

###  Split the data into training and testing sets

In [7]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 42)
regressor=DecisionTreeRegressor(random_state=42)
regressor.fit(X_train,y_train)

DecisionTreeRegressor(random_state=42)

### Training the model with the training data

In [8]:
y_pred=regressor.predict(X_test)

### Predict the model

In [9]:
df.head()

Team 1 Win  Team 2 Win  rating  rating-t  rating-ct  Average combat score  \
0           1           0    1.32      1.21       1.51                 267.0   
1           1           0    1.22      1.21       1.24                 239.0   
2           1           0    1.13      1.14       1.13                 202.0   
3           1           0    1.06      0.35       2.29                 267.0   
4           1           0    0.81      0.51       1.31                 105.0   

   kills   k-t  k-ct  deaths  ...  tkmd-ct  kills assists survive trade perct  \
0   20.0  11.0   9.0    11.0  ...      6.0                               0.68   
1   16.0  12.0   4.0    13.0  ...      0.0                               0.84   
2   12.0   9.0   3.0    10.0  ...      1.0                               0.68   
3   18.0   8.0  10.0    13.0  ...      8.0                               0.79   
4    8.0   4.0   4.0    12.0  ...      2.0                               0.74   

   average damage per round  headshot percentage  first kill  fk-t  fk-ct  \
0                     160.0                 0.31         1.0   0.0    1.0   
1                     158.0                 0.41         0.0   0.0    0.0   
2                     148.0                 0.33         2.0   2.0    0.0   
3                     158.0                 0.27         7.0   3.0    4.0   
4                      64.0                 0.39         0.0   0.0    0.0   

   first death  fd-t  fd-ct  
0          1.0   0.0    1.0  
1          1.0   1.0    0.0  
2          1.0   1.0    0.0  
3          5.0   5.0    0.0  
4          1.0   0.0    1.0  

[5 rows x 27 columns]

### Displaying the first 5 rows of the dataset in order to set up basic predictions, I also made a copy of my workbook in excel with the exact data to make predictions easier to impute into the dataset

In [10]:
regressor.predict([[1.32,1.21,1.51,267,20,11,9,11,8,3,1,1,0,9,3,6,0.68,160,0.31,1,0,1,1,1,0]])

C:\Users\Connor\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\sklearn\base.py:493: UserWarning: X does not have valid feature names, but DecisionTreeRegressor was fitted with feature names
  warnings.warn(


array([1.])

### Taken from the first row of the dataset to show the prediction of the model

In [11]:
regressor.predict([[1.34,1.51,1,314,20,14,6,11,7,4,1,1,0,9,7,2,.61,214,.35,0,0,0,2,1,1]])

C:\Users\Connor\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\sklearn\base.py:493: UserWarning: X does not have valid feature names, but DecisionTreeRegressor was fitted with feature names
  warnings.warn(


array([0.])

### Taken from the 32nd row of the dataset to show the prediction of the model

In [12]:
from sklearn.metrics import r2_score
print('R-Square',r2_score(y_test,y_pred))

R-Square -0.9748167863081605


### R-Square value of the model to determine whats going on with the model

# With a negative r-squared value, and the use of this model I wont be able to get any meaning from r^2. 
# It is possible not able to capture the non-linear relationship between the features and the target variable. So I will take a Mean Square error test instead to gather more info. 

In [13]:
from sklearn.metrics import mean_squared_error as msc
from math import sqrt
rms=sqrt(msc(y_test,y_pred))
print('The Mean Square error is', rms)

The Mean Square error is 0.6935630188191351


### The mean square error of the model to determine the "accuracy" of the model
### With the low score of .6936 and an r-square that is negative its possible this model could work but I will want to gather more info and test other models before finilizing anything.

# Chaning the model to predict kills overall instead of just purely a win or a lose. 

In [14]:
X = df[['rating', 'rating-t', 'rating-ct', 'Average combat score', 'deaths',
        'd-t', 'd-ct', 'assists', 'a-t', 'a-ct', 'total kills minus deaths', 'tkmd-t', 'tkmd-ct', 'kills assists survive trade perct', 'average damage per round', 'headshot percentage', 'first kill', 'fk-t', 'fk-ct', 'first death', 'fd-t', 'fd-ct']]

### Chaning X to specific columns I need to predict kills

In [15]:
y = df['kills']

### Setting my Y as new predictor in kills

In [16]:
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = 42)
regressor=DecisionTreeRegressor(random_state=42)
regressor.fit(X_train,y_train)

DecisionTreeRegressor(random_state=42)

### Retraining data

In [17]:
y_pred=regressor.predict(X_test)

### Predict the model

In [18]:
regressor.predict([[1.32,1.21,1.51,267,11,8,3,1,1,0,9,3,6,0.68,160,0.31,1,0,1,1,0,1]])

C:\Users\Connor\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\sklearn\base.py:493: UserWarning: X does not have valid feature names, but DecisionTreeRegressor was fitted with feature names
  warnings.warn(


array([20.])

### Testing Model

In [19]:
regressor.predict([[1.34,1.51,1,314,11,7,4,1,1,0,9,7,2,.61,214,.35,0,0,0,2,1,1]])

C:\Users\Connor\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\sklearn\base.py:493: UserWarning: X does not have valid feature names, but DecisionTreeRegressor was fitted with feature names
  warnings.warn(


array([20.])

### Testing Model

In [20]:
regressor.predict([[0.64,0.38,0.79,129,14,6,8,3,0,3,-5,-4,-1,0.68,87,0.27,1,0,1,0,0,0]])

C:\Users\Connor\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.10_qbz5n2kfra8p0\LocalCache\local-packages\Python310\site-packages\sklearn\base.py:493: UserWarning: X does not have valid feature names, but DecisionTreeRegressor was fitted with feature names
  warnings.warn(


array([9.])

### Testing Model

In [21]:
print('R-Square',r2_score(y_test,y_pred))

R-Square 0.9982068765623329


### New r^2 giving me almost a perfect score (Which isnt what i want normally a false positive)

In [22]:
rms=sqrt(msc(y_test,y_pred))
print('The Mean Square error is', rms)

The Mean Square error is 0.23459836628604097


### Super low MSE which tells me something is just wrong with this model because its too "perfect" 

# Final Thoughts


##### For the future of this project I am excited to build multiple more models and trying out different libaries inder order to build a predictive repitour of stats to display to consumer of the "app" or "program" I am making. 

# The 2nd model is basically telling me this is 100% perfect yet I do not trust something like this. There is potential in my first model but with more testing I might be able to find a better fit for models with this data.